drift/lib/src/runtime/query_builder/schema/column_impl.dart

```diff
@@ -93,10 +93,20 @@ class GeneratedColumn<T> extends Column<T> {
   /// [here](https://www.sqlite.org/syntax/column-def.html), into the given
   /// buffer.
   void writeColumnDefinition(GenerationContext into) {
-    into.buffer.write('$escapedName $typeName');
+    final isSerial = into.dialect == SqlDialect.postgres &&
+        type is IntType &&
+        _defaultConstraints == 'PRIMARY KEY AUTOINCREMENT';
```

Comment:
Checking for the generated default constraints feels a bit hacky to me. Maybe we could add a field like `bool isPrimaryKeyWithAutoIncrement = false` field to the column and have the generator set that to true when needed?

Reply:
Drift also use this hacky way ;)

```dart
/// Whether this column has an `AUTOINCREMENT` primary key constraint that was
  /// created by drift.
  bool get hasAutoIncrement =>
      _defaultConstraints?.contains('AUTOINCREMENT') == true;
```

Reply:
Touché, putting that into a getter is fine then :)